### PR TITLE
[logging] Several tweaks to #130

### PR DESCRIPTION
**Pull request by ejgallego**
See individual commits for details.

### CODE REVIEW

File: lsp/io.ml [hunk 1]
```diff
@@ -74,9 +74,11 @@ let read_request ic =
   | Invalid_argument msg -> raise (ReadError msg)
 
 let mut = Mutex.create ()
+let log = ref (fun _ _ -> ())
```
Comment (artagnon) [inline lsp/io.ml:79]: I'm not too fond of this hack: can you try `let ... and let` instead?

File: controller/coq_lsp.ml [hunk 2]
```diff
@@ -501,15 +500,15 @@ let rec process_queue ofmt ~state =
...
+    LIO.trace "process_queue" ("Serving Request: " ^ m);
     (* We let Coq work normally now *)
     Control.interrupt := false;
     dispatch_message ofmt ~state dict);
   process_queue ofmt ~state
 
 let lsp_cb oc =
   Fleche.Io.CallBack.
-    { log_error = LIO.log_error
+    { log_error = LIO.trace ?extra:None
```
Comment (artagnon) [inline controller/coq_lsp.ml:511]: Maybe rename this field to `trace_verbose`, and perhaps introduce a `trace_message`?

File: lsp/io.ml (hunk 1 above)
Comment (ejgallego) [reply lsp/io.ml:79]: Indeed that's a good idea, but I have another PR to improve some of the limitations of `trace` logging where this goes away.

File: controller/coq_lsp.ml (hunk 2 above)
Comment (ejgallego) [reply controller/coq_lsp.ml:511]: Renamed, I think just `trace` + extra info for verbose is ok.

**Pull request merged**